aiida_quantumespresso/parsers/pp.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+""" Parser for pp.x calculations """
+from __future__ import absolute_import
+
+import traceback
+
+import numpy as np
+
+from aiida import orm
+from aiida.common import exceptions
+
+from aiida_quantumespresso.calculations.pp import PpCalculation
+from aiida_quantumespresso.utils.mapping import get_logging_container
+from .base import Parser
+
+from six.moves import range
+from six.moves import zip
+
+class PpParser(Parser):
+    """
+    Parser for pp.x calculations
+    """
+
+    # Lookup for units
+    units_dict = {
+        0: 'e/bohr^3', # Electrons, electronic charge density
+        1: 'Ry', # Total potential
+        2: 'Ry', # Ionic potential
+        3: 'states/eV.bohr^3', # Density of states over an energy range
+        4: 'states/eV.bohr^3', # Local density of electronic entropy
+        5: 'states/eV.bohr^3', # Simulated STM images from LDOS
+        6: 'e/bohr^3', # Spin density
+        7: 'e/bohr^3', # WFN contribution to charge density, assuming collinear spins
+        8: '1', # Electron localization function, dimensionless
+        9: 'e/bohr^3', # Charge density minus superposition of atomic densities
+        10: 'states/bohr^3', # Integrated local density of states (ILDOS)
+        11: 'Ry', # Bare + hartree potential
+        12: 'Ry', # Electric field
```

Comment:
Here I think the unit is right, but the comment should be "the sawtooth electric field _potential_"

Reply:
Agreed